### ISSUE TITLE
cache API results

### ISSUE THREAD

**Issue by mtmail** [shayonj/emergency_number#3]
The full database is only 100kb. It could be included in the gem. Alternatively the gem could download the full database and cache it, either 24 hours or forever. That will ease the load on the "heavily rate limited" API.

**Comment by shayonj** [comment]
Yep. I was initially going to have the [data.json](https://github.com/EmergencyNumberAPI/data/blob/master/data.json) reside in the gem. But wanted to start off gradually :). I am researching on some good ways to cache it. In the next two releases I plan to 
- Cache (Priority)
- Introduce a command line utility (not a priority but a good feature to have :)

Thanks!

**Comment by mtmail** [comment]
Great gem.
The https://github.com/RubyMoney/eu_central_bank has nice example code for caching currency data.

**Comment by shayonj** [comment]
Fantastic. I am also researching to see if Redis will be a good fit. Might be a little overwhelming. Will update here soon.

cc @johnnye . Do you anticipate any concerns?

**Comment by johnnye** [comment]
"Heavily Rate Limited" - is my fault, a hangover from when I was thinking about some marketing ideas for it. The current rate limit is 5 request per second - and should be able to be upped with no concern moving forwards. I'd rather set expectations low and ramp up. 

I see no issues with caching the data, I am a little cautious about bundling the data with the gem as it could lead to stale data being shipped. However, _if_ the API had method to deliver a full set of results for clients to cache, that should solve this issue. 

Redis is fine for the job, you could put a TTL of 1 week or so on it as the data won't change that much.

**Comment by shayonj** [comment]
No worries. This is good learning exercise for me. Yeah bundling data as is in the gem, is probably the quickest solution but not the best. A smart combination of redis and giving users the ability to manually update is what I am thinking now.

Will update here around mid/end of the week.

**Comment by johnnye** [comment]
I realeased some code yesterday that includes `Expires` Headers for all API responses & also the full data set available from an endpoint. I need to update the changelog for this. Any questions - let me know.
